PDFlex/pdflex-backend
# LoginUseCaseTests

## Motivation and Context
Fully-functioning testing for our LoginUseCase. Serves as the template for future tests. Ready for Thursday's in-class standup presentation.

Tutorial used: https://spring.io/guides/gs/testing-web/

Jia Hao suggested we use JUnit for our use case interactor testing since we can "test our code without worrying about Spring Boot". However, I wanted something more sophisticated that actually tests our backend, database and their HTTP connection (POST, GET). So I am using the actual Spring testing.
Note by Jia Hao: "For unit testing a use case interactor, which depends on an external database connection and data, it might be worth it to mock the database data instead. Mockito is a good library for mocking stuff". I have found it sufficient to use our own database. Have not looked into Mockito.





## Your Changes


- Created LoginUseCaseTests
- Deleted FlexApplicationTests (created previously by Vithu)

**Type of change** (select all that apply):




- [x] Test update (change that modifies or updates tests only)

## Notes for Reviewer


In this branch (test-cases) I have already merged my LoginUseCase pull request into here.

I suggest you follow the [Project Individual Contribution](https://q.utoronto.ca/courses/314288/pages/project-individual-contribution?module_item_id=5179742#:~:text=Two%20examples%20of%20code%20reviews%20that%20you%20performed%20(1%25)) guide when creating your review so this can count towards your 1%!

How to run the test cases:
1. Navigate to flex > src > test > java > consonants.flex > LoginuseCaseTests
2. Run the file

The following should appear:
![image](https://github.com/PDFlex/pdflex-backend/assets/99620362/8a69bf53-3fa4-4ff9-aba9-44f707242cca)

## Checklist
